src/doppler/main.go

```diff
@@ -163,11 +176,12 @@ func StartHeartbeats(localIp string, ttl time.Duration, config *config.Config, l
 		return
 	}
 
-	adapter := StoreAdapterProvider(config.EtcdUrls, config.EtcdMaxConcurrentRequests)
-	adapter.Connect()
+	if storeAdapter == nil {
```

Comment:
Is it possible for `storeAdapter` to be `nil` here? Given your implementation of `NewStoreAdapter` on [line 48](https://github.com/cloudfoundry/loggregator/pull/50/files#diff-274dbfc3a53f7e4827ec278d9ea767a9R48), it seems impossible to arrive here without `storeAdapter` in a working state.